pretyping/indrec.ml

```diff
@@ -637,23 +637,26 @@ let check_elim env sigma (kn, n) mib u lrecspec =
 
 (* Check all the blocks are mutual, and not given twice *)
 let check_mut env sigma (kn, n) mib u lrecspec =
-  List.fold_left (fun ln ((kni, ni),dep,s) ->
-    (* Check al the block are mutual  *)
-    if not (QMutInd.equal env kn kni)
-    then raise (RecursionSchemeError (env, NotMutualInScheme ((kn, n),(kni, ni))));
-    (* Check none is given twice *)
-    if Int.List.mem ni ln
-    then raise (RecursionSchemeError (env, DuplicateInductiveBlock (kn, ni)))
-    else ni::ln
-  ) [n] lrecspec
+  let _ : int list =
+    List.fold_left (fun ln ((kni, ni),dep,s) ->
+        (* Check all the blocks are mutual  *)
+        if not (QMutInd.equal env kn kni)
+        then raise (RecursionSchemeError (env, NotMutualInScheme ((kn, n),(kni, ni))));
+        (* Check none is given twice *)
+        if Int.List.mem ni ln
```

Comment:
While you're at it this could have been extremely prematurely optimized and replaced by a set...